src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -54,6 +55,9 @@ internal partial class RequestExecutionQueue
         private readonly CancellationTokenSource _cancelSource;
         private readonly DocumentChangeTracker _documentChangeTracker;
 
+        private readonly Dictionary<Workspace, Solution> _lastWorkspaceSolution = new();
+        private readonly Dictionary<Workspace, Solution> _lastLSPSolution = new();
```

Comment:
These will go once servers are per-workspace, and just be two `Solution` fields

Reply:
1. any threading concerns with these?  or do we know all messages are one at a time?
2. probably good to doc these (esp. how the two dicts differ).

Reply:
any concerns with holding onto the workspace here as well?  I guess once we switch to server per workspace it's moot then

Reply:
I docced this a bit, can expand in the follow up when things are per workspace, but I suspect it will be a bit more obvious then anyway.

Reply:
> any concerns with holding onto the workspace here as well? I guess once we switch to server per workspace it's moot then

Yeah, I expect this to be here for a matter of days at most. If I wasn't busy this afternoon with .NET Conf I probably wouldn't have bothered with it at all.